### ISSUE TITLE
API Endpoint Request

### ISSUE THREAD

**Issue by sgoggins** [chaoss/augur#327]
#### Calculate Contributions and Display Developers Contributing n % of all Commits (Repo and Repo Group) 
We want to build a pie chart that shows the commit share of the developers "inside the bus factor", and then provides a slice, as well, for "all other committers". 
1. To orient ourselves, first, how many total commits are there? 
```sql
    select sum(patches) from (select repo_group_id, email, year, patches from dm_repo_group_annual
    where year = 2017
    order by patches desc)a; 
```
For our example, there are 9,671 patches in a repository group of 112 repositories.

2.  What is the distribution of committers and total commits? 
```sql
    SELECT
        repo_group_id,
        email,
        SUM ( patches ) as commitsyear 
    FROM
        ( SELECT repo_group_id, email, YEAR, patches FROM dm_repo_group_annual WHERE YEAR = 2017 ORDER BY patches DESC )  a
    GROUP BY
        repo_group_id,
        email
        order by commitsyear desc; 
```
There are 783 committers in our sample. The top committer has 650 commits, the bottom committer has 1 commit, and there are 411 such individuals. 

3. How many committers are in the set that contributes 50% of the 9,671 commits. 
    - We know the total commits are 9,671, so we need to loop through committers in an ordered list, from the top down, until we arrive at the list of top contributors who contribute 50% of the total commits. In this case,  `1/2*9,671 = 4,835.5`, and we round up to 4,836. 
    - When we loop through top committers, keeping a cumulative total, we add each developer to the list until we exceed 4,836, at which point we have the list. There are 30 committers in our list (plus one row for everyone not in the "developers with the most contributions whose total commits add up to 50% (or other percent) of the total): 
```
        Developer           Commits in a Year
  
```
4. Our pie chart would then have 30 developers, and a 31st slice for "all other contributors". This is a large case, obviously. Perhaps a percentile that is lower would be appropriate. This is parameterizable. 
**Pseudo Code: 
```python
    commits =1; 
    for i in all_committers: 
        commits = commits+developerCommits
        developerlist = add developer to list; 
        if commits > totalcommits/2 
            break; 
```
The end result we want is a list with the commits per developer until we reach 1/2 or some other share of the total commits PLUS, one slice for ALL the other commits made by people outside the group whose commit total, counted from the most commits down, adds up to that percentage (50% or otherwise). 

5. The JSON we would want to return probably looks like: 
```json
    [
        {
            "user_id": 1,
            "email": "s@goggins.com",
            "commits": 333,
            "repo_group_name": "rails",
            "repo_group_id": 21000
        },
        {
            "user_id": 2,
            "email": "d@goggins.com",
            "commits": 331,
            "repo_group_name": "rails",
            "repo_group_id": 21000
        },
        {
            "user_id": 1,
            "email": "other_contributors",
            "commits": 133,
            "repo_group_name": "rails",
            "repo_group_id": 21000
        }
    ]

```

**Comment by parthsharma2** [comment]
@sgoggins  what do we name this?

**Comment by parthsharma2** [comment]
Implemented in #330

**Comment by Nebrethar** [comment]
@sgoggins Are you okay with closing this issue?
